## Post-Process Information
# bag_name: "/home/mmmfarrell/magicc/roscopter_ws/src/bags/groot_sensors_stationary.bag"
bag_name: "/home/mmmfarrell/magicc/roscopter_ws/src/bags/groot_sensors_triangle.bag"
status_topic: "/status"
imu_topic: "/imu/data"
pose_topic: ""
odom_topic: ""
gnss_topic: ""
ublox_gnss_topic: "/PosVelEcef"
inertial_sense_gnss_topic: "/ins/gps"

start_time: 10
duration: 75

## Is_flying checks
enable_arm_check: false # require armed before checking for is_flying
is_flying_threshold: 10.5 # if accel measurement exceed this magnitude, set is_flying

## Logging
log_prefix: "/tmp/roscopter/ekf/"
enable_log: true

## Measurement Processing
enable_out_of_order: false
use_truth: true # Whether or not to use mocap measurements
use_gnss: true
use_alt: true
use_zero_vel: true

R_zero_vel: [0.001, 0.001, 0.001, # vel
              0.001] # Altitude

## Initial Uncertainty
P0: [10., 10.0, 10.0, # pos
     0.1, 0.1, 2.0, # att
     0.001, 0.001, 0.001, # vel
     0.3, 0.3, 0.3, # accel bias
     0.1, 0.1, 0.1, # gyro bias
     100.0] # reference altitude

## Additive Process Noise
Qx: [0.000, 0.000, 0.000,       # pos
     0.001, 0.001, 0.001,       # att
     0.001, 0.001, 0.001,       # vel
     0.001, 0.001, 0.001,       # accel bias
     0.0001, 0.0001, 0.0001,    # gyro bias
     0.0001]                    # reference altitude

# Constant Parameeters
p_b2g: [0, 0, 0]

# Initial State
manual_ref_lla: false # Use manual ref_lla or use first lla from gps
ref_lla: [40.24632, -111.646882, 1415.0]
ref_heading: 0.0 # reference heading for inertial frame (from north)

x0: [0, 0, 0,       # pos
     1, 0, 0, 0]    # att

## Sensor Noise
accel_noise_stdev: 0.5
gyro_noise_stdev: 0.02
position_noise_stdev: 0.01
attitude_noise_stdev: 0.01
alt_noise_stdev: 0.1
